schema/crdb/add-instance-boot-device/up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE omicron.public.instance ADD COLUMN IF NOT EXISTS boot_device STRING(63);
```

Comment:
if an instance has only one disk attached it seems reasonable to use that as the value of `boot_device` here. i have no idea how to write sql that would produce that outcome though. would that be a second `up02.sql` that's something like...

```
UPDATE (select instance.id, instance.boot_device, disk.id) FROM instance 
    JOIN disk ON instance.id=disk.attach_instance_id
    HAVING COUNT(disk.id) == 1)
    SET instance.boot_device=disk.id;
```
?

Reply:
I think that would be a separate migration. Claude suggests something more like this:

```sql
UPDATE instance
  SET boot_device = disk.id
  FROM instance
  JOIN disk ON instance.id = disk.attach_instance_id
  WHERE instance.id IN (
    SELECT attach_instance_id
    FROM disk
    GROUP BY attach_instance_id
    HAVING COUNT(*) = 1
  );
```

Reply:
We probably want to exclude instances that are already deleted with a `time_deleted is NULL` condition?

Reply:
Never mind - deleted instance should have the `attach_instance_id` reset so they won't be touched as it appears.

Reply:
fwiw this ended up looking like [up02.sql](https://github.com/oxidecomputer/omicron/pull/6585/files#diff-da4c1e6160e13f959047827393955c2989f94dbdec580cdc3c2bbdc2a4628f50)

notable wrinkles being:
* i added `WHERE disk_state = 'attached' AND attach_instance_id IS NOT NULL` because i'm extremely paranoid about accidentally catching a disk incorrectly. these should be redundant with `instance.id = d.attach_instance_id` - if `attach_instance_id` is null then that will not keep the disk anyway. but, paranoia.
* `min(id) as id` because even though `HAVING COUNT(*) = 1`, every selected column needs to be either in GROUP BY or run through an aggregate function. `min({one_thing})=one_thing` so it's just gross, not risky.

(sqlite allows unaggregated columns to be named but the result if more than one row is grouped is undefined, as i learned a few months ago)

anyway good job Claude